ignite/metrics/metric.py

```diff
@@ -526,7 +526,7 @@ def __setstate__(self, d: Dict) -> None:
         self.__dict__.update(d)
 
 
-def sync_all_reduce(*attrs: Any) -> Callable:
+def sync_all_reduce(*attrs: Any, op: str = "SUM") -> Callable:
```

Comment:
Ok, that is a way to do that. But we could want mix different ops
```python
@sync_all_reduce("sum_var", "max_var")
def compute(self):
    pass   
```

What do you think about the following suggestion ?
```python
@sync_all_reduce("sum_var", ("max_var", "MAX"))
def compute(self):
    pass   
```
I used a tuple, it is not very friendly but the idea is to provide a way to select the op to apply per variable. Maybe we could introduce specific tools:
```python
@sync_all_reduce("sum_var", max_reduced_variable("max_var"))
def compute(self):
    pass   
```
Interested by your opinion.

Reply:
The way i updated `sync_all_reduce` will allow minimal changes and that was my concern, i think chaining decorators with different `op` arg will solve the "select the op to apply per variable".

Reply:
We maydo it with several decorators ?
```python
@sync_all_reduce("sum_var")
@sync_all_reduce("max_var", op="MAX")
def compute(self):
    pass
```
And internally, resolve the decoration in a bit more sophisticated way...

Reply:
Thanks for the explanation @vfdev-5, that's what i mean by 
> chaining decorators with different `op` arg will solve the "select the op to apply per variable" issue.

Reply:
However, chaining decorators may look nice, i'm not a fan of multiple subfunc calls: each decorator brings 2 new functions. With 2 sync_all_reduce, we have 4 functions to call before calling `compute`. I'd like to avoid that...

Reply:
Maybe, Sylvain's solution like `@sync_all_reduce("sum_var", ("max_var", "MAX"))` would be the cleanest one which we can simplify a bit more as 
```
@sync_all_reduce("sum_var", "max_var:max")
def compute(self):
    pass   
```

Reply:
I can do that, should i make `@sync_all_reduce("sum_var", ("max_var", "MAX"))` tuples or what?

Reply:
Personnally, I prefer `@sync_all_reduce("sum_var", "max_var:max", "min_var:min", "another_var:sum")`.
@sdesrozis what do you think ?

Reply:
I prefer encode the op in the str rather than handle tuples 😉 It is very elegant.